src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php

```diff
@@ -274,8 +278,11 @@ private function getDocBlockFromMethod(string $class, string $ucFirstProperty, i
                 }
 
                 if (
-                    (self::ACCESSOR === $type && 0 === $reflectionMethod->getNumberOfRequiredParameters())
-                    || (self::MUTATOR === $type && $reflectionMethod->getNumberOfParameters() >= 1)
+                    $this->canAccessMemberBasedOnItsVisibility($reflectionMethod)
+                    && (
+                        (self::ACCESSOR === $type && 0 === $reflectionMethod->getNumberOfRequiredParameters())
+                        || (self::MUTATOR === $type && $reflectionMethod->getNumberOfParameters() >= 1)
+                    )
```

Comment:
```suggestion
                    (
                        (self::ACCESSOR === $type && 0 === $reflectionMethod->getNumberOfRequiredParameters())
                        || (self::MUTATOR === $type && $reflectionMethod->getNumberOfParameters() >= 1)
                    )
                    && $this->canAccessMemberBasedOnItsVisibility($reflectionMethod)
```

To prevent a useless function call in the happy path.